lib/pages/trade/DesktopTrade.dart

```diff
@@ -33,6 +36,15 @@ class _DesktopTradeState extends State<DesktopTrade> {
     _tokenToInputController.dispose();
   }
 
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+
+    canTrade = context.watch<BuyDialogBloc>().state.canTrade;
```

Comment:
This should be removed.

Please use a `BlocBuilder`/`context.watch` inside `build` method if need to build some widget or `BlocListener` if you need to react with side effects like showing a dialog or navigate.